src/api/c/moddims.cpp

```diff
@@ -91,21 +94,61 @@ af_err af_moddims(af_array *out, const af_array in,
     return AF_SUCCESS;
 }
 
+template<typename T>
+Array<T> flatten(const Array<T>& in)
+{
+    const dim4 newDims(in.elements());
+    return modDims<T>(in, newDims);
+}
+
+template Array<float  > flatten(const Array<float  >&);
+template Array<double > flatten(const Array<double >&);
+template Array<cfloat > flatten(const Array<cfloat >&);
+template Array<cdouble> flatten(const Array<cdouble>&);
+template Array<int    > flatten(const Array<int    >&);
+template Array<uint   > flatten(const Array<uint   >&);
+template Array<intl   > flatten(const Array<intl   >&);
+template Array<uintl  > flatten(const Array<uintl  >&);
+template Array<short  > flatten(const Array<short  >&);
+template Array<ushort > flatten(const Array<ushort >&);
+template Array<uchar  > flatten(const Array<uchar  >&);
+template Array<char   > flatten(const Array<char   >&);
+
+template<typename T>
+inline af_array flatten(const af_array in)
```

Comment:
Add these functions in anonymous namespace to prevent their use outside this file.